### PR TITLE
Add feature #4902, the option to edit extratext field (subtitle,summa…

### PR DESCRIPTION
**Pull request by pablozg**
…ry or description) and the episode field, because sometimes the recordings don't have the proper order (one missed, no free adapter, etc) and is good to have the option to add the "Season XX.Episode XX" text by hand if it not set previously.

### CODE REVIEW

File: src/dvr/dvr_db.c [hunk 1]
```diff
@@ -3230,6 +3230,43 @@ dvr_entry_class_disp_extratext_get(void *o)
   return &prop_ptr;
 }
 
+static int
+dvr_entry_class_disp_extratext_set(void *o, const void *v)
+{
+  dvr_entry_t *de = (dvr_entry_t *)o;
+  const char *lang = idnode_lang(o);
+  const char *s = "";
+  int field = 0;
```
Comment (perexg) [inline src/dvr/dvr_db.c]: It should be probably 1 as default (to edit the empty extra text).

File: src/dvr/dvr_db.c [hunk 2]
```diff
@@ -3230,6 +3230,43 @@ dvr_entry_class_disp_extratext_get(void *o)
...
+  v = tvh_str_default(v, "UnknownExtraText");
+  if (de->de_subtitle){
+    s = lang_str_get(de->de_subtitle, lang);
```
Comment (perexg) [inline src/dvr/dvr_db.c]: The lang_str_set() already returns 0 (no string change) or 1 (string changed), so you can do 'return lang_str_set(de->de_substitle, v, lang);' here.

File: src/dvr/dvr_db.c [hunk 3]
```diff
@@ -3230,6 +3230,43 @@ dvr_entry_class_disp_extratext_get(void *o)
...
 }
 
+static int
+dvr_entry_class_disp_extratext_set(void *o, const void *v)
+{
+  dvr_entry_t *de = (dvr_entry_t *)o;
+  const char *lang = idnode_lang(o);
+  const char *s = "";
+  int field = 0;
+  v = tvh_str_default(v, "UnknownExtraText");
```
Comment (perexg) [inline src/dvr/dvr_db.c]: Use a more universal (no international dependency) string here, like '?????'.

File: src/dvr/dvr_db.c [hunk 4]
```diff
@@ -3967,7 +4005,7 @@ const idclass_t dvr_entry_class = {
       .desc     = N_("Episode number/ID."),
       .set      = dvr_entry_class_disp_episode_set,
       .get      = dvr_entry_class_disp_episode_get,
-      .opts     = PO_RDONLY | PO_HIDDEN | PO_NOSAVE,
+      .opts     = PO_HIDDEN | PO_NOSAVE | PO_ADVANCED,
```
Comment (perexg) [inline src/dvr/dvr_db.c]: Why this was changed to 'advanced'?

File: src/dvr/dvr_db.c [hunk 5]
```diff
@@ -3715,7 +3752,8 @@ const idclass_t dvr_entry_class = {
       .name     = N_("Extra text"),
       .desc     = N_("Subtitle, summary or description of the program (if any) (display only)."),
       .get      = dvr_entry_class_disp_extratext_get,
-      .opts     = PO_RDONLY | PO_NOSAVE,
+      .set      = dvr_entry_class_disp_extratext_set,
+      .opts     = PO_NOSAVE | PO_ADVANCED,
```
Comment (perexg) [inline src/dvr/dvr_db.c]: I don't think that this should be advanced.

File: src/dvr/dvr_db.c [hunk 6]
```diff
@@ -3230,6 +3230,27 @@ dvr_entry_class_disp_extratext_get(void *o)
...
 }
 
+static int
+dvr_entry_class_disp_extratext_set(void *o, const void *v)
+{
+  dvr_entry_t *de = (dvr_entry_t *)o;
+  const char *lang = idnode_lang(o);
+  v = tvh_str_default(v, "?????");
+
+  if (de->de_subtitle && lang_str_get(de->de_subtitle, lang)[0] != '\0'){
```
Comment (pablozg) [inline src/dvr/dvr_db.c]: I don't know if this is the best method to do the compare "lang_str_get(de->de_subtitle, lang)[0]" but it works.

**Comment by perexg** [comment]
Merged. Thanks.

**Pull request closed without merging**